### ISSUE TITLE
negation of SC2252, `[ "$1" = foo ] && [ "$1" = bar ]`, is not detected

### ISSUE THREAD

**Issue by e-kwsm** [koalaman/shellcheck#3180]
#### For bugs
- Rule Id (if any, e.g. SC1000): [SC2252](https://www.shellcheck.net/wiki/SC2252)?
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# ! { [ "$1" != foo ] || [ "$1" != bar ]; }  # SC2252 is issued
if [ "$1" = foo ] && [ "$1" = bar ]
then
  echo "$1 is foo or bar"
fi
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

```
Line 3:
if [ "$1" = foo ] && [ "$1" = bar ]
                  ^-- SC2252 (warning): You probably wanted || here, otherwise it's always false.
```
